chain/chain/src/store.rs

```diff
@@ -1921,6 +1939,7 @@ impl<'a> ChainStoreUpdate<'a> {
     ) -> Result<(), Error> {
         let mut store_update = self.store().store_update();
         let header = self.get_block_header(&block_hash).expect("block header must exist").clone();
+        let next_epoch_id = header.epoch_id();
```

Comment:
Is this intended (instead of `header.next_epoch_id()`)?

Reply:
Absolutely. On the Canonical chain we replace the current hash with previous one below.